Polyhedron/demo/Polyhedron/MainWindow.h

```diff
@@ -17,12 +17,13 @@
 #include <QStringList>
 #include <QSet>
 #include <QModelIndex>
+#include <QLineEdit>
+#include <QWidgetAction>
+
 class Scene;
 class Viewer;
 class QTreeView;
 class QMenu;
-class QLineEdit;
-class QWidgetAction;
```

Comment:
Actually, for `QWidgetAction`, the declaration was sufficient, because this class only stores a pointer to it, whereas it stores a `QLineEdit` as a member (not a pointer to..).